After 24h no Temprature update with Climate Control

Hello,

i have the Beta 5 release addon running.
My BSH Gateway thing says it is online and my five ClimateControl things are online. 
After i add the BSH Thing all works fine. 
The setpoint_temprature and temprature Item become a automatics updates. After i change the value in the Bosch App the setpoint item in OH change his value under one second.
After 24h i can change a value and wait for temprature updates but nothing happen.

When i pause the BSH Thing and start it again i can change the values in the Bosch app and all works fine for 24 hours.
I made this yesterday and today. In my Log is no error. Yesterday i made the resart at 13:00 today at 14:00. I thing tomorrow at 14:00 i will get no updates.

I am from north of Hamburg. I can describe this better in German when you want.

Jörg

I got once a logged exception regarding "long polling". Perhaps the stack trace supports analysis

	[WARN ] [devices.bridge.BoschSHCBridgeHandler] - Long polling failed
	org.openhab.binding.boschshc.internal.exceptions.LongPollingFailedException: Unexpected exception during long polling request
		at org.openhab.binding.boschshc.internal.devices.bridge.LongPolling$1.onComplete(LongPolling.java:152) [bundleFile:?]
		at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:198) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:190) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:444) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.HttpReceiver.abort(HttpReceiver.java:527) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.HttpChannel.abortResponse(HttpChannel.java:156) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.HttpChannel.abort(HttpChannel.java:149) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.HttpExchange.abort(HttpExchange.java:257) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.HttpConversation.abort(HttpConversation.java:149) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.HttpRequest.abort(HttpRequest.java:768) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.TimeoutCompleteListener.onTimeoutExpired(TimeoutCompleteListener.java:51) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.io.CyclicTimeout$Wakeup.run(CyclicTimeout.java:282) [bundleFile:9.4.20.v20190813]
		at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
		at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
		at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
		at java.lang.Thread.run(Thread.java:834) [?:?]
	Caused by: java.util.concurrent.TimeoutException: Total timeout 30000 ms elapsed
		... 8 more

Oh no, it froze again. This must have happened just minutes after my post above. :crying_cat_face: 

![Screenshot from 2021-01-13 07-45-51](https://user-images.githubusercontent.com/12135716/104416122-853e8880-5573-11eb-8620-8c18d5945435.png)

The "unfreeze" and restart of the binding was on 2021-01-11 at ~ 19:30. It "froze" again at 2021-01-12 20:00.

Is it because the buggy and the new version of the binding are installed at the same time? How do I remove the old version? Removing it from the *openhab/addons* directory did not help.

```
232 │ Active │  80 │ 3.0.0.202012261941      │ org.openhab.binding.boschshc
233 │ Active │  80 │ 3.1.0.202101110957      │ org.openhab.binding.boschshc
```

This is how my *openhab.log* ends:

```
2021-01-12 19:57:53.652 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Got update for hdm:ZigBee:000d6XXXXXXXXX09
2021-01-12 19:57:53.654 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: hdm:ZigBee:000d6f000cb95099 - looking for hdm:ZigBee:000d6XXXXXXXXX09
2021-01-12 19:57:53.657 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: hdm:ZigBee:000d6XXXXXXXXX09 - looking for hdm:ZigBee:000d6XXXXXXXXX09
2021-01-12 19:57:53.659 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Found child: org.openhab.binding.boschshc.internal.devices.twinguard.BoschTwinguardHandler@ce4aa3 - calling processUpdate with {"temperatureRating":"GOOD","humidityRating":"GOOD","purity":890,"comfortZone":{"maxTemperature":25,"minTemperature":20,"custom":false,"name":"HALLWAY","maxHumidity":60,"minHumidity":40,"maxPurity":1000},"@type":"airQualityLevelState","purityRating":"GOOD","temperature":22.69,"description":"OK","humidity":44.31,"combinedRating":"GOOD"}
2021-01-12 19:57:53.662 [DEBUG] [ices.twinguard.BoschTwinguardHandler] - Twinguard: received update: AirQualityLevel {"temperatureRating":"GOOD","humidityRating":"GOOD","purity":890,"comfortZone":{"maxTemperature":25,"minTemperature":20,"custom":false,"name":"HALLWAY","maxHumidity":60,"minHumidity":40,"maxPurity":1000},"@type":"airQualityLevelState","purityRating":"GOOD","temperature":22.69,"description":"OK","humidity":44.31,"combinedRating":"GOOD"}
2021-01-12 19:57:53.665 [DEBUG] [ices.twinguard.BoschTwinguardHandler] - Parsed switch state of hdm:ZigBee:000d6XXXXXXXXX09: org.openhab.binding.boschshc.internal.devices.twinguard.dto.AirQualityLevelState@1393a2c
2021-01-12 19:57:53.682 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: hdm:ZigBee:000d6XXXXXXXXX3c - looking for hdm:ZigBee:000d6XXXXXXXXX09
2021-01-12 19:57:53.686 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: hdm:ZigBee:000d6XXXXXXXXXfd - looking for hdm:ZigBee:000d6XXXXXXXXX09
2021-01-12 19:57:53.689 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: hdm:ZigBee:000d6XXXXXXXXX2c - looking for hdm:ZigBee:000d6XXXXXXXXX09
2021-01-12 19:57:53.692 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-12 19:58:13.714 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-12 19:58:13.717 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-12 19:58:33.778 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-12 19:58:33.780 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-12 19:58:53.848 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-12 19:58:53.853 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-12 19:59:03.008 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-12 19:59:03.011 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-12 19:59:03.078 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"jsonrpc":"2.0","error":{"code":-32001,"message":"No subscription with id: eeb57n4ke-33"}}

2021-01-12 20:12:22.141 [WARN ] [nternal.phonebook.Tr064PhonebookImpl] - Failed to get phonebook with URL https://192.168.74.1:49443/phonebook.lua?sid=06c0cda484fee8c6&pbid=0:
java.util.concurrent.TimeoutException: Total timeout 2000 ms elapsed
	at org.eclipse.jetty.client.TimeoutCompleteListener.onTimeoutExpired(TimeoutCompleteListener.java:51) ~[?:?]
	at org.eclipse.jetty.io.CyclicTimeout$Wakeup.run(CyclicTimeout.java:282) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2021-01-13 07:14:13.728 [ERROR] [d4j.internal.RRD4jPersistenceService] - Could not create rrd4j database file '/openhab/userdata/persistence/rrd4j/SBBathroom_CurrentPlayingTime.rrd': null
```

@coeing I fear this issue needs to be reopened. I uninstalled the old binding and restarted OH. I think this time it ran for ~36 hours and now it is frozen again. I think @tuxianerDE reported the same in https://github.com/stefan-kaestle/openhab2-addons/issues/66#issuecomment-760046514 .

![image](https://user-images.githubusercontent.com/12135716/104691411-79370000-5706-11eb-8da3-f0f6aad076c6.png)

In the bundle list there is only one instance of the *boschshc* binding.

```
233 │ Active │  80 │ 3.1.0.202101110957      │ org.openhab.binding.boschshc
```

This is now my *openhab.log* ends. Give me a shout when you want more from the log.

```
2021-01-14 19:34:23.328 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-14 19:34:23.330 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 19:34:43.401 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-14 19:34:43.408 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 19:35:03.478 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-14 19:35:03.482 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 19:35:23.548 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-14 19:35:23.550 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 19:35:43.616 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-14 19:35:43.621 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 19:36:03.688 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-14 19:36:03.694 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 19:36:23.758 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-14 19:36:23.760 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 19:36:33.209 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[{"path":"/devices/hdm:ZigBee:000d6XXXXXXXXX3c/services/AirQualityLevel","@type":"DeviceServiceData","id":"AirQualityLevel","state":{"temperatureRating":"GOOD","humidityRating":"GOOD","purity":810,"comfortZone":{"maxTemperature":25,"minTemperature":20,"custom":false,"name":"KIDS","maxHumidity":60,"minHumidity":40,"maxPurity":1000},"@type":"airQualityLevelState","purityRating":"GOOD","temperature":22.34,"description":"OK","humidity":45.37,"combinedRating":"GOOD"},"deviceId":"hdm:ZigBee:000d6XXXXXXXXX3c"}],"jsonrpc":"2.0"}

2021-01-14 19:36:33.212 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Got update for hdm:ZigBee:000d6XXXXXXXXX3c
2021-01-14 19:36:33.214 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: hdm:ZigBee:000d6XXXXXXXXX99 - looking for hdm:ZigBee:000d6XXXXXXXXX3c
2021-01-14 19:36:33.217 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: hdm:ZigBee:000d6XXXXXXXXX09 - looking for hdm:ZigBee:000d6XXXXXXXXX3c
2021-01-14 19:36:33.219 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: hdm:ZigBee:000d6XXXXXXXXX3c - looking for hdm:ZigBee:000d6XXXXXXXXX3c
2021-01-14 19:36:33.221 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Found child: org.openhab.binding.boschshc.internal.devices.twinguard.BoschTwinguardHandler@1cd2d6f - calling processUpdate with {"temperatureRating":"GOOD","humidityRating":"GOOD","purity":810,"comfortZone":{"maxTemperature":25,"minTemperature":20,"custom":false,"name":"KIDS","maxHumidity":60,"minHumidity":40,"maxPurity":1000},"@type":"airQualityLevelState","purityRating":"GOOD","temperature":22.34,"description":"OK","humidity":45.37,"combinedRating":"GOOD"}
2021-01-14 19:36:33.223 [DEBUG] [ices.twinguard.BoschTwinguardHandler] - Twinguard: received update: AirQualityLevel {"temperatureRating":"GOOD","humidityRating":"GOOD","purity":810,"comfortZone":{"maxTemperature":25,"minTemperature":20,"custom":false,"name":"KIDS","maxHumidity":60,"minHumidity":40,"maxPurity":1000},"@type":"airQualityLevelState","purityRating":"GOOD","temperature":22.34,"description":"OK","humidity":45.37,"combinedRating":"GOOD"}
2021-01-14 19:36:33.226 [DEBUG] [ices.twinguard.BoschTwinguardHandler] - Parsed switch state of hdm:ZigBee:000d6XXXXXXXXX3c: org.openhab.binding.boschshc.internal.devices.twinguard.dto.AirQualityLevelState@1731de2
2021-01-14 19:36:33.243 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: hdm:ZigBee:000d6XXXXXXXXXfd - looking for hdm:ZigBee:000d6XXXXXXXXX3c
2021-01-14 19:36:33.246 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: hdm:ZigBee:000d6XXXXXXXXX2c - looking for hdm:ZigBee:000d6XXXXXXXXX3c
2021-01-14 19:36:33.249 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 19:36:33.378 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-14 19:36:33.380 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 19:36:33.448 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"jsonrpc":"2.0","error":{"code":-32001,"message":"No subscription with id: eeb57n4ke-37"}}

2021-01-14 19:36:33.450 [WARN ] [.internal.devices.bridge.LongPolling] - Long poll response contained no results: {"jsonrpc":"2.0","error":{"code":-32001,"message":"No subscription with id: eeb57n4ke-37"}}

2021-01-14 19:36:33.460 [WARN ] [.internal.devices.bridge.LongPolling] - Got long poll error: No subscription with id: eeb57n4ke-37 (code: -32001)
2021-01-14 19:36:33.462 [WARN ] [.internal.devices.bridge.LongPolling] - Subscription eeb57n4ke-37 became invalid, subscribing again
2021-01-14 19:36:33.465 [DEBUG] [.internal.devices.bridge.LongPolling] - Subscribe: Sending request: {"jsonrpc":"2.0","method":"RE/subscribe","params":["com/bosch/sh/remote/*",null]} - using httpClient BoschHttpClient@5fb4e6{STARTED}
2021-01-15 00:00:59.538 [INFO ] [ng.tr064.internal.soap.SOAPConnector] - Failed to get Tr064ChannelConfig{channelType=wanAccessType, getAction=GetCommonLinkProperties, dataType='string, parameter='null'}: java.util.concurrent.TimeoutException: Total timeout 2000 ms elapsed
2021-01-15 06:26:09.976 [ERROR] [d4j.internal.RRD4jPersistenceService] - Could not create rrd4j database file '/openhab/userdata/persistence/rrd4j/SBDining_CurrentPlayingTime.rrd': null
```

Hi,
i have the same problem. I think it is still after 24 hours.

Here are my Log:
`
2021-01-14 11:11:10.308 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Got update for hdm:HomeMaticIP:3014F711A000005BB85E1B04
2021-01-14 11:11:10.309 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_2 - looking for hdm:HomeMaticIP:3014F711A000005BB85E1B04
2021-01-14 11:11:10.310 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_1 - looking for hdm:HomeMaticIP:3014F711A000005BB85E1B04
2021-01-14 11:11:10.311 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_4 - looking for hdm:HomeMaticIP:3014F711A000005BB85E1B04
2021-01-14 11:11:10.311 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_5 - looking for hdm:HomeMaticIP:3014F711A000005BB85E1B04
2021-01-14 11:11:10.312 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_3 - looking for hdm:HomeMaticIP:3014F711A000005BB85E1B04
2021-01-14 11:11:10.313 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 11:11:10.379 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[{"path":"/devices/roomClimateControl_hz_4/services/TemperatureLevel","@type":"DeviceServiceData","id":"TemperatureLevel","state":{"@type":"temperatureLevelState","temperature":14.3},"deviceId":"roomClimateControl_hz_4"}],"jsonrpc":"2.0"}

2021-01-14 11:11:10.380 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Got update for roomClimateControl_hz_4
2021-01-14 11:11:10.381 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_2 - looking for roomClimateControl_hz_4
2021-01-14 11:11:10.382 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_1 - looking for roomClimateControl_hz_4
2021-01-14 11:11:10.383 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_4 - looking for roomClimateControl_hz_4
2021-01-14 11:11:10.383 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Found child: org.openhab.binding.boschshc.internal.devices.climatecontrol.ClimateControlHandler@75d087e0 - calling processUpdate with {"@type":"temperatureLevelState","temperature":14.3}
2021-01-14 11:11:10.385 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_5 - looking for roomClimateControl_hz_4
2021-01-14 11:11:10.387 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_3 - looking for roomClimateControl_hz_4
2021-01-14 11:11:10.387 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 11:11:30.451 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-14 11:11:30.452 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 11:11:36.151 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[{"path":"/devices/hdm:HomeMaticIP:3014F711A000005BB85E1B0B/services/TemperatureLevel","@type":"DeviceServiceData","id":"TemperatureLevel","state":{"@type":"temperatureLevelState","temperature":17.5},"deviceId":"hdm:HomeMaticIP:3014F711A000005BB85E1B0B"}],"jsonrpc":"2.0"}

2021-01-14 11:11:36.175 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Got update for hdm:HomeMaticIP:3014F711A000005BB85E1B0B
2021-01-14 11:11:36.175 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_2 - looking for hdm:HomeMaticIP:3014F711A000005BB85E1B0B
2021-01-14 11:11:36.176 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_1 - looking for hdm:HomeMaticIP:3014F711A000005BB85E1B0B
2021-01-14 11:11:36.176 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_4 - looking for hdm:HomeMaticIP:3014F711A000005BB85E1B0B
2021-01-14 11:11:36.176 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_5 - looking for hdm:HomeMaticIP:3014F711A000005BB85E1B0B
2021-01-14 11:11:36.176 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_3 - looking for hdm:HomeMaticIP:3014F711A000005BB85E1B0B
2021-01-14 11:11:36.176 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 11:11:36.247 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[{"path":"/devices/roomClimateControl_hz_3/services/TemperatureLevel","@type":"DeviceServiceData","id":"TemperatureLevel","state":{"@type":"temperatureLevelState","temperature":17.5},"deviceId":"roomClimateControl_hz_3"}],"jsonrpc":"2.0"}

2021-01-14 11:11:36.248 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Got update for roomClimateControl_hz_3
2021-01-14 11:11:36.249 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_2 - looking for roomClimateControl_hz_3
2021-01-14 11:11:36.249 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_1 - looking for roomClimateControl_hz_3
2021-01-14 11:11:36.250 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_4 - looking for roomClimateControl_hz_3
2021-01-14 11:11:36.250 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_5 - looking for roomClimateControl_hz_3
2021-01-14 11:11:36.251 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Registered device: roomClimateControl_hz_3 - looking for roomClimateControl_hz_3
2021-01-14 11:11:36.251 [DEBUG] [devices.bridge.BoschSHCBridgeHandler] - Found child: org.openhab.binding.boschshc.internal.devices.climatecontrol.ClimateControlHandler@714b463d - calling processUpdate with {"@type":"temperatureLevelState","temperature":17.5}
2021-01-14 11:11:36.253 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 11:11:56.311 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-14 11:11:56.325 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 11:12:06.631 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"result":[],"jsonrpc":"2.0"}

2021-01-14 11:12:06.646 [DEBUG] [.internal.devices.bridge.LongPolling] - Sending long poll request
2021-01-14 11:12:06.703 [DEBUG] [.internal.devices.bridge.LongPolling] - Long poll response: {"jsonrpc":"2.0","error":{"code":-32001,"message":"No subscription with id: eebi62mc3-251"}}

2021-01-14 11:12:06.707 [WARN ] [.internal.devices.bridge.LongPolling] - Long poll response contained no results: {"jsonrpc":"2.0","error":{"code":-32001,"message":"No subscription with id: eebi62mc3-251"}}

2021-01-14 11:12:06.717 [WARN ] [.internal.devices.bridge.LongPolling] - Got long poll error: No subscription with id: eebi62mc3-251 (code: -32001)
2021-01-14 11:12:06.717 [WARN ] [.internal.devices.bridge.LongPolling] - Subscription eebi62mc3-251 became invalid, subscribing again
2021-01-14 11:12:06.718 [DEBUG] [.internal.devices.bridge.LongPolling] - Subscribe: Sending request: {"jsonrpc":"2.0","method":"RE/subscribe","params":["com/bosch/sh/remote/*",null]} - using httpClient BoschHttpClient@4756fa7b{STARTED}
2021-01-14 11:12:16.728 [WARN ] [devices.bridge.BoschSHCBridgeHandler] - Long polling failed
org.openhab.binding.boschshc.internal.exceptions.LongPollingFailedException: Error on subscribe request
	at org.openhab.binding.boschshc.internal.devices.bridge.LongPolling.subscribe(LongPolling.java:120) ~[bundleFile:?]
	at org.openhab.binding.boschshc.internal.devices.bridge.LongPolling.onLongPollResponse(LongPolling.java:195) [bundleFile:?]
	at org.openhab.binding.boschshc.internal.devices.bridge.LongPolling$1.onComplete(LongPolling.java:162) [bundleFile:?]
	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:198) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:190) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:444) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.client.HttpReceiver.responseSuccess(HttpReceiver.java:390) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.messageComplete(HttpReceiverOverHTTP.java:316) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.http.HttpParser.parseFields(HttpParser.java:1139) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1498) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.parse(HttpReceiverOverHTTP.java:172) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:135) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:73) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:133) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:154) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onFillable(SslConnection.java:426) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:320) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:158) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:367) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782) [bundleFile:9.4.20.v20190813]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:918) [bundleFile:9.4.20.v20190813]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.util.concurrent.TimeoutException: Total timeout 10000 ms elapsed
	at org.eclipse.jetty.client.TimeoutCompleteListener.onTimeoutExpired(TimeoutCompleteListener.java:51) ~[?:?]
	at org.eclipse.jetty.io.CyclicTimeout$Wakeup.run(CyclicTimeout.java:282) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	... 1 more
2021-01-14 20:10:55.248 [WARN ] [e.internal.SseItemStatesEventBuilder] - Attempting to send a state update of an item which doesn't exist: undefined
2021-01-14 20:12:02.705 [WARN ] [e.internal.SseItemStatesEventBuilder] - Attempting to send a state update of an item which doesn't exist: undefined
2021-01-14 20:31:19.120 [WARN ] [e.internal.SseItemStatesEventBuilder] - Attempting to send a state update of an item which doesn't exist: undefined
2021-01-14 20:31:20.258 [WARN ] [e.internal.SseItemStatesEventBuilder] - Attempting to send a state update of an item which doesn't exist: undefined
2021-01-14 20:31:22.985 [WARN ] [e.internal.SseItemStatesEventBuilder] - Attempting to send a state update of an item which doesn't exist: undefined
2021-01-14 20:34:59.852 [WARN ] [e.internal.SseItemStatesEventBuilder] - Attempting to send a state update of an item which doesn't exist: undefined
2021-01-14 20:41:31.997 [WARN ] [e.internal.SseItemStatesEventBuilder] - Attempting to send a state update of an item which doesn't exist: undefined
2021-01-15 11:50:01.071 [WARN ] [e.internal.SseItemStatesEventBuilder] - Attempting to send a state update of an item which doesn't exist: undefined
2021-01-15 17:37:16.069 [WARN ] [e.internal.SseItemStatesEventBuilder] - Attempting to send a state update of an item which doesn't exist: undefined
`

Here is my trace from yesterday

	2021-01-19 20:56:57.467 [WARN ] [.internal.devices.bridge.LongPolling] - Long poll response contained no results: {"jsonrpc":"2.0","error":{"code":-32001,"message":"No subscription with id: ef5lmcknf-1"}}
	2021-01-19 20:56:57.474 [WARN ] [.internal.devices.bridge.LongPolling] - Got long poll error: No subscription with id: ef5lmcknf-1 (code: -32001)
	2021-01-19 20:56:57.475 [WARN ] [.internal.devices.bridge.LongPolling] - Subscription ef5lmcknf-1 became invalid, subscribing again
	2021-01-19 20:57:07.480 [WARN ] [devices.bridge.BoschSHCBridgeHandler] - Long polling failed
	org.openhab.binding.boschshc.internal.exceptions.LongPollingFailedException: Error on subscribe request
		at org.openhab.binding.boschshc.internal.devices.bridge.LongPolling.subscribe(LongPolling.java:120) ~[bundleFile:?]
		at org.openhab.binding.boschshc.internal.devices.bridge.LongPolling.onLongPollResponse(LongPolling.java:195) [bundleFile:?]
		at org.openhab.binding.boschshc.internal.devices.bridge.LongPolling$1.onComplete(LongPolling.java:162) [bundleFile:?]
		at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:198) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:190) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:444) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.HttpReceiver.responseSuccess(HttpReceiver.java:390) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.messageComplete(HttpReceiverOverHTTP.java:316) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.http.HttpParser.parseFields(HttpParser.java:1139) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1498) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.parse(HttpReceiverOverHTTP.java:172) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:135) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:73) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:133) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:154) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onFillable(SslConnection.java:426) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:320) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:158) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:367) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782) [bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:918) [bundleFile:9.4.20.v20190813]
		at java.lang.Thread.run(Thread.java:834) [?:?]
	Caused by: java.util.concurrent.TimeoutException: Total timeout 10000 ms elapsed
		at org.eclipse.jetty.client.TimeoutCompleteListener.onTimeoutExpired(TimeoutCompleteListener.java:51) ~[bundleFile:9.4.20.v20190813]
		at org.eclipse.jetty.io.CyclicTimeout$Wakeup.run(CyclicTimeout.java:282) ~[bundleFile:9.4.20.v20190813]
		at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
		at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
		at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
		... 1 more

resulting in a "COMMUNICATION_ERROR"
![Screenshot_Openhab_20210119-222803](https://user-images.githubusercontent.com/1284770/105136674-56379200-5af2-11eb-9c79-9a92bcc2a052.png)

--- a/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschHttpClient.java
+++ b/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschHttpClient.java
@@ -194,6 +194,8 @@ public class BoschHttpClient extends HttpClient {
      * @return created HTTP request instance
      */
     public Request createRequest(String url, HttpMethod method, @Nullable Object content) {
+        logger.trace(String.format("Create request for http client %s", this.toString()));
+
         Request request = this.newRequest(url).method(method).header("Content-Type", "application/json");
         if (content != null) {
             String body = GSON.toJson(content);
@@ -220,9 +222,11 @@ public class BoschHttpClient extends HttpClient {
      */
     public <TContent> TContent sendRequest(Request request, Class<TContent> responseContentClass)
             throws InterruptedException, TimeoutException, ExecutionException {
+        logger.trace("Send request: {}", request.toString());
+
         ContentResponse contentResponse = request.send();
 
-        logger.debug("BoschHttpClient: response complete: {} - return code: {}", contentResponse.getContentAsString(),
+        logger.debug("Received response: {} - status: {}", contentResponse.getContentAsString(),
                 contentResponse.getStatus());
 
         try {

--- a/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschHttpClient.java
+++ b/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschHttpClient.java
@@ -194,7 +194,7 @@ public class BoschHttpClient extends HttpClient {
      * @return created HTTP request instance
      */
     public Request createRequest(String url, HttpMethod method, @Nullable Object content) {
-        logger.trace(String.format("Create request for http client %s", this.toString()));
+        logger.trace("Create request for http client {}", this.toString());
 
         Request request = this.newRequest(url).method(method).header("Content-Type", "application/json");
         if (content != null) {

--- a/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschSHCBridgeHandler.java
+++ b/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschSHCBridgeHandler.java
@@ -291,12 +291,12 @@ public class BoschSHCBridgeHandler extends BaseBridgeHandler {
         BoschHttpClient httpClient = this.httpClient;
         if (httpClient == null) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
-                    "Long polling failed and could not be restarted because http client is null");
+                    "@text/offline.long-polling-failed.http-client-null");
             return;
         }
 
         this.updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.UNKNOWN.NONE,
-                String.format("Long polling failed, will try to reconnect: %s", e.getMessage()));
+                String.format("@text/offline.long-polling-failed.trying-to-reconnect", e.getMessage()));
         scheduleInitialAccess(httpClient);
     }
 

--- a/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschSHCBridgeHandler.java
+++ b/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschSHCBridgeHandler.java
@@ -287,8 +287,20 @@ public class BoschSHCBridgeHandler extends BaseBridgeHandler {
     }
 
     private void handleLongPollFailure(Throwable e) {
-        logger.warn("Long polling failed", e);
-        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, "Long polling failed");
+        logger.warn("Long polling failed, trying to restart", e);
+        BoschHttpClient httpClient = this.httpClient;
+        if (httpClient == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Long polling failed and could not be restarted because http client is null");
+            return;
+        }
+
+        try {
+            this.longPolling.start(httpClient);
+        } catch (LongPollingFailedException restartException) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    String.format("Long polling failed and could not be restarted: %s", restartException.getMessage()));
+        }
     }
 
     /**

--- a/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschSHCBridgeHandler.java
+++ b/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschSHCBridgeHandler.java
@@ -287,7 +287,7 @@ public class BoschSHCBridgeHandler extends BaseBridgeHandler {
     }
 
     private void handleLongPollFailure(Throwable e) {
-        logger.warn("Long polling failed, trying to restart", e);
+        logger.warn("Long polling failed, will try to reconnect", e);
         BoschHttpClient httpClient = this.httpClient;
         if (httpClient == null) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
@@ -295,12 +295,9 @@ public class BoschSHCBridgeHandler extends BaseBridgeHandler {
             return;
         }
 
-        try {
-            this.longPolling.start(httpClient);
-        } catch (LongPollingFailedException restartException) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
-                    String.format("Long polling failed and could not be restarted: %s", restartException.getMessage()));
-        }
+        this.updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.UNKNOWN.NONE,
+                String.format("Long polling failed, will try to reconnect: %s", e.getMessage()));
+        scheduleInitialAccess(httpClient);
     }
 
     /**

--- a/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java
+++ b/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java
@@ -122,10 +122,6 @@ public class LongPolling {
         }
     }
 
-    private void executeResubscribe(BoschHttpClient httpClient) {
-        scheduler.execute(() -> this.resubscribe(httpClient));
-    }
-
     /**
      * Create a new subscription for long polling.
      * 
@@ -164,23 +160,10 @@ public class LongPolling {
         request.send(new BufferingResponseListener() {
             @Override
             public void onComplete(@Nullable Result result) {
-                Throwable failure = result != null ? result.getFailure() : null;
-                if (failure != null) {
-                    if (failure instanceof ExecutionException) {
-                        if (failure.getCause() instanceof AbortLongPolling) {
-                            logger.debug("Canceling long polling for subscription id {} because it was aborted",
-                                    subscriptionId);
-                        } else {
-                            longPolling.handleFailure.accept(new LongPollingFailedException(
-                                    "Unexpected exception during long polling request", failure));
-                        }
-                    } else {
-                        longPolling.handleFailure.accept(new LongPollingFailedException(
-                                "Unexpected exception during long polling request", failure));
-                    }
-                } else {
-                    longPolling.onLongPollResponse(httpClient, subscriptionId, this.getContentAsString());
-                }
+                // NOTE: This handler runs inside the HTTP thread, so we schedule the response handling in a new thread
+                // because the HTTP thread is terminated after the timeout expires.
+                scheduler.execute(() -> longPolling.onLongPollComplete(httpClient, subscriptionId, result,
+                        this.getContentAsString()));
             }
         });
     }
@@ -188,47 +171,62 @@ public class LongPolling {
     /**
      * This is the handler for responses of long poll requests.
      * 
-     * @implNote Please be aware that this function runs inside the HTTP request thread, so be careful with calling
-     *           other functions that might have to run on a main thread
-     * 
      * @param httpClient HTTP client which received the response
      * @param subscriptionId Id of subscription the response is for
+     * @param result Complete result of the response
      * @param content Content of the response
      */
-    private void onLongPollResponse(BoschHttpClient httpClient, String subscriptionId, String content) {
+    private void onLongPollComplete(BoschHttpClient httpClient, String subscriptionId, @Nullable Result result,
+            String content) {
         // Check if thing is still online
         if (this.aborted) {
             logger.debug("Canceling long polling for subscription id {} because it was aborted", subscriptionId);
             return;
         }
 
-        logger.debug("Long poll response: {}", content);
-
-        String nextSubscriptionId = subscriptionId;
-
-        LongPollResult longPollResult = gson.fromJson(content, LongPollResult.class);
-        if (longPollResult != null && longPollResult.result != null) {
-            this.handleResult.accept(longPollResult);
+        Throwable failure = result != null ? result.getFailure() : null;
+        if (failure != null) {
+            if (failure instanceof ExecutionException) {
+                if (failure.getCause() instanceof AbortLongPolling) {
+                    logger.debug("Canceling long polling for subscription id {} because it was aborted",
+                            subscriptionId);
+                } else {
+                    this.handleFailure.accept(new LongPollingFailedException(
+                            "Unexpected exception during long polling request", failure));
+                }
+            } else {
+                this.handleFailure.accept(
+                        new LongPollingFailedException("Unexpected exception during long polling request", failure));
+            }
         } else {
-            logger.warn("Long poll response contained no results: {}", content);
+            logger.debug("Long poll response: {}", content);
 
-            // Check if we got a proper result from the SHC
-            LongPollError longPollError = gson.fromJson(content, LongPollError.class);
+            String nextSubscriptionId = subscriptionId;
 
-            if (longPollError != null && longPollError.error != null) {
-                logger.warn("Got long poll error: {} (code: {})", longPollError.error.message,
-                        longPollError.error.code);
+            LongPollResult longPollResult = gson.fromJson(content, LongPollResult.class);
+            if (longPollResult != null && longPollResult.result != null) {
+                this.handleResult.accept(longPollResult);
+            } else {
+                logger.warn("Long poll response contained no results: {}", content);
 
-                if (longPollError.error.code == LongPollError.SUBSCRIPTION_INVALID) {
-                    logger.warn("Subscription {} became invalid, subscribing again", subscriptionId);
-                    this.executeResubscribe(httpClient);
-                    return;
+                // Check if we got a proper result from the SHC
+                LongPollError longPollError = gson.fromJson(content, LongPollError.class);
+
+                if (longPollError != null && longPollError.error != null) {
+                    logger.warn("Got long poll error: {} (code: {})", longPollError.error.message,
+                            longPollError.error.code);
+
+                    if (longPollError.error.code == LongPollError.SUBSCRIPTION_INVALID) {
+                        logger.warn("Subscription {} became invalid, subscribing again", subscriptionId);
+                        this.resubscribe(httpClient);
+                        return;
+                    }
                 }
             }
-        }
 
-        // Execute next run.
-        this.executeLongPoll(httpClient, nextSubscriptionId);
+            // Execute next run.
+            this.longPoll(httpClient, nextSubscriptionId);
+        }
     }
 
     @SuppressWarnings("serial")

--- a/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java
+++ b/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java
@@ -122,6 +122,25 @@ public class LongPolling {
         }
     }
 
+    private void executeResubscribe(BoschHttpClient httpClient) {
+        scheduler.execute(() -> this.resubscribe(httpClient));
+    }
+
+    /**
+     * Create a new subscription for long polling.
+     * 
+     * @param httpClient Http client to send requests to
+     */
+    private void resubscribe(BoschHttpClient httpClient) {
+        try {
+            String subscriptionId = this.subscribe(httpClient);
+            this.executeLongPoll(httpClient, subscriptionId);
+        } catch (LongPollingFailedException e) {
+            this.handleFailure.accept(e);
+            return;
+        }
+    }
+
     private void executeLongPoll(BoschHttpClient httpClient, String subscriptionId) {
         scheduler.execute(() -> this.longPoll(httpClient, subscriptionId));
     }
@@ -166,6 +185,16 @@ public class LongPolling {
         });
     }
 
+    /**
+     * This is the handler for responses of long poll requests.
+     * 
+     * @implNote Please be aware that this function runs inside the HTTP request thread, so be careful with calling
+     *           other functions that might have to run on a main thread
+     * 
+     * @param httpClient HTTP client which received the response
+     * @param subscriptionId Id of subscription the response is for
+     * @param content Content of the response
+     */
     private void onLongPollResponse(BoschHttpClient httpClient, String subscriptionId, String content) {
         // Check if thing is still online
         if (this.aborted) {
@@ -192,12 +221,8 @@ public class LongPolling {
 
                 if (longPollError.error.code == LongPollError.SUBSCRIPTION_INVALID) {
                     logger.warn("Subscription {} became invalid, subscribing again", subscriptionId);
-                    try {
-                        nextSubscriptionId = this.subscribe(httpClient);
-                    } catch (LongPollingFailedException e) {
-                        this.handleFailure.accept(e);
-                        return;
-                    }
+                    this.executeResubscribe(httpClient);
+                    return;
                 }
             }
         }

--- a/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java
+++ b/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java
@@ -117,7 +117,8 @@ public class LongPolling {
             String subscriptionId = response.getResult();
             return subscriptionId;
         } catch (TimeoutException | ExecutionException | InterruptedException e) {
-            throw new LongPollingFailedException("Error on subscribe request", e);
+            throw new LongPollingFailedException(
+                    String.format("Error on subscribe request (Http client: %s)", httpClient.toString()), e);
         }
     }
 

--- a/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java
+++ b/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java
@@ -118,7 +118,8 @@ public class LongPolling {
             return subscriptionId;
         } catch (TimeoutException | ExecutionException | InterruptedException e) {
             throw new LongPollingFailedException(
-                    String.format("Error on subscribe request (Http client: %s)", httpClient.toString()), e);
+                    String.format("Error on subscribe (Http client: %s): %s", httpClient.toString(), e.getMessage()),
+                    e);
         }
     }
 

--- a/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java
+++ b/bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java
@@ -185,6 +185,7 @@ public class LongPolling {
             return;
         }
 
+        // Check if response was failure or success
         Throwable failure = result != null ? result.getFailure() : null;
         if (failure != null) {
             if (failure instanceof ExecutionException) {
@@ -208,24 +209,24 @@ public class LongPolling {
             if (longPollResult != null && longPollResult.result != null) {
                 this.handleResult.accept(longPollResult);
             } else {
-                logger.warn("Long poll response contained no results: {}", content);
+                logger.debug("Long poll response contained no result: {}", content);
 
                 // Check if we got a proper result from the SHC
                 LongPollError longPollError = gson.fromJson(content, LongPollError.class);
 
                 if (longPollError != null && longPollError.error != null) {
-                    logger.warn("Got long poll error: {} (code: {})", longPollError.error.message,
+                    logger.debug("Got long poll error: {} (code: {})", longPollError.error.message,
                             longPollError.error.code);
 
                     if (longPollError.error.code == LongPollError.SUBSCRIPTION_INVALID) {
-                        logger.warn("Subscription {} became invalid, subscribing again", subscriptionId);
+                        logger.debug("Subscription {} became invalid, subscribing again", subscriptionId);
                         this.resubscribe(httpClient);
                         return;
                     }
                 }
             }
 
-            // Execute next run.
+            // Execute next run
             this.longPoll(httpClient, nextSubscriptionId);
         }
     }

--- a/bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/i18n/boschshc.properties
+++ b/bundles/org.openhab.binding.boschshc/src/main/resources/OH-INF/i18n/boschshc.properties
@@ -3,3 +3,5 @@
 offline.conf-error-pairing = Press pairing button on the Bosch Smart Home Controller.
 offline.not-reachable = Smart Home Controller is not reachable.
 offline.conf-error-ssl = The SSL connection to the Bosch Smart Home Controller is not possible.
+offline.long-polling-failed.http-client-null = Long polling failed and could not be restarted because http client is null
+offline.long-polling-failed.trying-to-reconnect = Long polling failed, will try to reconnect: %s

